### ISSUE TITLE
/archive/data-2.0/ might be unmounting in bad ways...

### ISSUE THREAD

**Issue by edickie** [TIGRLab/admin#519]
Do the run script this morning (2016-04-27) is madness - 
half way through it says that nothing in data-2.0 can be read...
I this maybe an issue with the new storage?

**Comment by josephdviviano** [comment]
That sounds bad... I'll have a look.

**Comment by josephdviviano** [comment]
Permissions in data-2.0 ARE madness.

```
jdv@davinci:/archive/data-2.0$ ls -al
total 180
drwxrwxr-x 1 nobody tigrlab      756 Mar 31 10:53 .
dr-xr-xr-x 1 root   root         152 Jan 22 14:35 ..
drwxrwxr-x 1 nobody 4294967294   272 Oct 19  2015 ANDT
drwxrwxr-x 1 nobody 4294967294   272 Feb 22 15:24 ASDD
drwxrwxr-x 1 nobody dawn         148 Aug  6  2015 assets
drwxrwxr-x 1 nobody 4294967294   228 Mar 30 11:09 code
drwxrwxr-x 1 nobody 4294967294   292 Nov 13 19:23 COGBDO
drwxrwxr-x 1 nobody 4294967294   292 Nov 13 19:23 COGBDY
drwxrwxr-x 1 nobody 4294967294   360 Mar 24 12:56 DBDC
drwxrwxr-x 1 nobody 4294967294   260 Apr 16 22:06 DTI
drwxrwxr-x 1 nobody 4294967294   412 Mar 29 11:09 .git
-rw-rw-r-- 1 nobody dawn         126 Aug 14  2015 .gitignore
drwxrwxr-x 1 nobody 4294967294   156 May 14  2015 HCP
-rw-rw-r-- 1 nobody       1007 62921 Feb 18 22:09 .jobscript
drwxrwxrwx 1   1001 tigrlab     4096 Apr 27 08:00 logs
drwxrwxr-x 1 nobody 4294967294   184 Nov  1 15:05 OBI_POND
drwxrwxr-x 1 nobody 4294967294   208 Nov  1 15:05 OCDMEG
drwxrwxr-x 1 nobody 4294967294   292 May 14  2015 PACTMD
drwxrwxr-x 1 nobody 4294967294   204 Nov  1 15:06 PASS
drwxrwxr-x 1 nobody 4294967294   372 Mar  9 10:47 PRELAPSE
-rw-rw-r-- 1 nobody dawn        2224 Mar 31 10:45 README.md
drwxrwxr-x 1 nobody 4294967294   292 Oct 16  2015 RTMSWM
drwxrwxr-x 1 nobody 4294967294   364 Mar  7 16:51 SPINS
drwxrwxr-x 1 nobody 4294967294   348 Dec 18 10:53 STOPPD
drwxrwxr-x 1 nobody 4294967294   204 Mar  8 15:28 SWM
drwxr-xr-x 1 nobody 4294967294   224 Sep  9  2015 UNITTEST
drwxrwxr-x 1 nobody 4294967294   244 Feb  4 15:08 VIPR
drwxrwxr-x 1 nobody 4294967294   244 Aug 18  2015 VRMD
drwxrwxr-x 1 nobody       1022   728 May 15  2015 WEBSITE
drwxrwxr-x 1 nobody       1022   172 Apr 22  2015 XNAT
```

**Comment by pipitone** [comment]
Interesting. Okay, there are two things that may be at play here: a) `nfs_automount` is supposed to remount `/mnt/tigrlab` if there are connection issues - so we need to see what has been happening in `/var/log/syslog` wrt that because the run log contains "`No such file or directory`" errors, and b) clearly NFS isn't doing user/group lookups properly from the IPA server.

**Comment by josephdviviano** [comment]
Hi @pipitone I'm trying to run the reset permissions bit of `run.sh` now.

**Comment by pipitone** [comment]
@josephdviviano That's likely not the problem since the permissions look reasonable on other machines.

**Comment by pipitone** [comment]
The fact that *nobody* is showing up as the user means NFS didn't correctly lookup the user info.

**Comment by josephdviviano** [comment]
Oh? Why would they be different on tigrsrv and other machines?

PS we had a remount this morning:
```
Apr 27 08:18:56 tigrsrv logger: nfs_automount: [CRIT] (dataset 1) Remote server/NFS service at 'rprshnas01.camh.ca' unreachable; confirming unmounted status of share '/tigrlab/tigrlab'.
Apr 27 08:19:56 tigrsrv logger: nfs_automount: [CRIT] (dataset 1) Remote NFS share 'rprshnas01.camh.ca:/tigrlab/tigrlab' is unmounted; attempting mount.
```

**Comment by josephdviviano** [comment]
All the logs in `/var/log/sssd/` are empty on tigrsrv.

**Comment by josephdviviano** [comment]
put this line
```
Domain = camhres.ca
```
/etc/idmapd.conf

**Comment by josephdviviano** [comment]
@pipitone I could use some help with this.

**Comment by josephdviviano** [comment]
This is what I know:

Ever since I changed `idmapd`, we got a bunch of lines like this in `/var/log/syslog`

    Apr 27 16:25:15 tigrsrv [sssd[ldap_child[6250]]]: Failed to initialize credentials using keytab [default]: Cannot find KDC for realm "CAMHRES.CA". Unable to create GSSAPI-encrypted LDAP connection.

If you look here: `vi /etc/krb5.conf`:

```
#File modified by ipa-client-install                                            
                                                                                
includedir /var/lib/sss/pubconf/krb5.include.d/                                 
                                                                                
[libdefaults]                                                                   
  default_realm = CAMHRES.CA                                                    
  dns_lookup_realm = true                                                       
  dns_lookup_kdc = true                                                         
  rdns = false                                                                  
  ticket_lifetime = 24h                                                         
  forwardable = yes                                                             
                                                                                
[realms]                                                                        
  CAMHRES.CA = {                                                                
    pkinit_anchors = FILE:/etc/ipa/ca.crt                                       
  }                                                                             
                                                                                
[domain_realm]                                                                  
  .camhres.ca = CAMHRES.CA                                                      
  camhres.ca = CAMHRES.CA 
```

I wonder if our kerberos config is in conflict with the one set up on `rprshnas01.camh.ca`

**Comment by josephdviviano** [comment]
Also, when I set the verbosity of `idmapd` and restarted it, this happened:

```
22473 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11855]]]: Failed to initialize credentials using keytab [default]: Cannot find KDC for realm "CAMHRES.CA". Unable to create GSSAPI-encrypted LDAP connection.
22474 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11855]]]: Cannot find KDC for realm "CAMHRES.CA"
22475 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11856]]]: Failed to initialize credentials using keytab [default]: Cannot find KDC for realm "CAMHRES.CA". Unable to create GSSAPI-encrypted LDAP connection.
22476 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11856]]]: Cannot find KDC for realm "CAMHRES.CA"
22477 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11857]]]: Failed to initialize credentials using keytab [default]: Cannot find KDC for realm "CAMHRES.CA". Unable to create GSSAPI-encrypted LDAP connection.
22478 Apr 27 16:39:59 tigrsrv [sssd[ldap_child[11857]]]: Cannot find KDC for realm "CAMHRES.CA"
22479 Apr 27 16:40:00 tigrsrv dhclient: DHCPDISCOVER on eth3 to 255.255.255.255 port 67 interval 6 (xid=0x2461564)
22480 Apr 27 16:40:00 tigrsrv ntpd[5111]: Listen normally on 7748 eth3 fe80::ca1f:66ff:fee5:e718 UDP 123
22481 Apr 27 16:40:00 tigrsrv ntpd[5111]: peers refreshed                             
22482 Apr 27 16:40:00 tigrsrv ntpd[5111]: new interface(s) found: waking up resolver  
22483 Apr 27 16:40:06 tigrsrv dhclient: DHCPDISCOVER on eth3 to 255.255.255.255 port 67 interval 14 (xid=0x2461564)
22484 Apr 27 16:40:17 tigrsrv NetworkManager[17942]: <info> (eth3): IP6 addrconf timed out or failed.
22485 Apr 27 16:40:17 tigrsrv NetworkManager[17942]: <info> Activation (eth3) Stage 4 of 5 (IPv6 Configure Timeout) scheduled...
22486 Apr 27 16:40:17 tigrsrv NetworkManager[17942]: <info> Activation (eth3) Stage 4 of 5 (IPv6 Configure Timeout) started...
22487 Apr 27 16:40:17 tigrsrv NetworkManager[17942]: <info> Activation (eth3) Stage 4 of 5 (IPv6 Configure Timeout) complete.
22488 Apr 27 16:40:18 tigrsrv kernel: [885862.200855] init: idmapd main process (48308) killed by TERM signal
22489 Apr 27 16:40:19 tigrsrv rpc.idmapd[11884]: libnfsidmap: using (default) domain: camhres.ca
22490 Apr 27 16:40:19 tigrsrv rpc.idmapd[11884]: libnfsidmap: Realms list: 'CAMHRES.CA'
22491 Apr 27 16:40:19 tigrsrv rpc.idmapd[11884]: libnfsidmap: loaded plugin /lib/x86_64-linux-gnu/libnfsidmap/nsswitch.so for method nsswitch
22492 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: Expiration time is 600 seconds.      
22493 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: Opened /proc/net/rpc/nfs4.nametoid/channel
22494 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: Opened /proc/net/rpc/nfs4.idtoname/channel
22495 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: New client: 37a                      
22496 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: New client: 37b                      
22497 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: New client: 381                      
22498 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: Opened /run/rpc_pipefs/nfs/clnt381/idmap
22499 Apr 27 16:40:19 tigrsrv rpc.idmapd[11885]: New client: 3b1  
```

Note:  `22488 Apr 27 16:40:18 tigrsrv kernel: [885862.200855] init: idmapd main process (48308) killed by TERM signal`

**Comment by pipitone** [comment]
sssd wasn't started, and when I attempted to start it failed immediately. I re-enrolled tigrsrv with IPA (enrolling is done via `ipa-client` role in ansible): 

```
$ sudo ipa-client-install --uninstall 
$ sudo ipa-client-install --enable-dns-updates --force-ntpd --mkhomedir --force-join
$ sudo service sssd status
sssd start/running, process 27146
```

This hasn't changed resolving the user/groups on `/archive/data-2.0`.

**Comment by josephdviviano** [comment]
Hey @pipitone 

Looks like a reboot cleared things up on both davinci and my machine (davinci). I've removed all references to `/archive/data-2.0` in the fstab on both machines... `nfs_automount` is the only device that is mounting this now.

Of note, I also had to manually shut down `tigrsrv` due to it hanging on `nfs`, and kick `sssd` on reboot (permissions of this file were incorrect on tigrsrv as well). I'll be watching things but the issue appears to be resolved.
